ConsoleApplication8.cpp

```diff
@@ -65,44 +101,74 @@ class RemoteControl {
     }
 };
 
+// Перечисление для команд
+enum CommandType {
+    TURN_ON,
+    TURN_OFF,
+    CHANGE_COLOR_RED,
+    CHANGE_COLOR_GREEN,
+    CHANGE_COLOR_BLUE,
+    EXIT
+};
+
 int main() {
     setlocale(LC_ALL, "Russian");
     Lamp livingRoomLamp;
 
     // Создаем команды для управления лампой
     std::shared_ptr<Command> turnOnLampCommand = std::make_shared<TurnOnCommand>(livingRoomLamp);
     std::shared_ptr<Command> turnOffLampCommand = std::make_shared<TurnOffCommand>(livingRoomLamp);
+    std::shared_ptr<Command> changeColorRedCommand = std::make_shared<ChangeColorCommand>(livingRoomLamp, "красный");
+    std::shared_ptr<Command> changeColorGreenCommand = std::make_shared<ChangeColorCommand>(livingRoomLamp, "зеленый");
+    std::shared_ptr<Command> changeColorBlueCommand = std::make_shared<ChangeColorCommand>(livingRoomLamp, "синий");
 
     // Создаем пульт управления и добавляем команды
     RemoteControl remoteControl;
     remoteControl.addCommand(turnOnLampCommand);
     remoteControl.addCommand(turnOffLampCommand);
+    remoteControl.addCommand(changeColorRedCommand);
+    remoteControl.addCommand(changeColorGreenCommand);
+    remoteControl.addCommand(changeColorBlueCommand);
 
     // Интерактивное меню для выбора команд через консоль
     int choice;
     do {
         std::cout << "Выберите команду:" << std::endl;
-        std::cout << "1. Включить лампу" << std::endl;
-        std::cout << "2. Выключить лампу" << std::endl;
-        std::cout << "0. Выйти" << std::endl;
+        std::cout << "0. Включить лампу" << std::endl;
+        std::cout << "1. Выключить лампу" << std::endl;
+        std::cout << "2. Изменить цвет лампы на красный" << std::endl;
+        std::cout << "3. Изменить цвет лампы на зеленый" << std::endl;
+        std::cout << "4. Изменить цвет лампы на синий" << std::endl;
+        std::cout << "5. Выйти" << std::endl;
         std::cout << "Ваш выбор: ";
         std::cin >> choice;
 
-        switch (choice) {
-        case 1:
+        CommandType command = static_cast<CommandType>(choice);
+
+        switch (command) {
```

Comment:
Я бы этот `switch` переписал так:
```suggestion
switch (command) {
        case TURN_ON:
        case TURN_OFF:
        case CHANGE_COLOR_RED:
        case CHANGE_COLOR_GREEN:
        case CHANGE_COLOR_BLUE:
        case EXIT:
            remoteControl.pressButton(command);
            break;
        default:
            std::cout << "Некорректный выбор. Попробуйте снова." << std::endl;
            break;
        }
```
Компактнее, надежнее, без дублирования кода. И энумератор используется полноценно, лишних "циферок" больше нет.

Не призываю исправлять, просто имей в виду такой вариант.